File: frontend/types/creative-task.ts
export type CreativeTaskStatus =
  | "pending"
  | "ready"
  | "in_progress"
  | "blocked"
  | "in_review"
  | "done"
  | "cancelled";

export interface CreativeTask {
  id: string;
  design_request_id: string;
  parent_task_id?: string | null;
  task_type: string;
  title: string;
  description?: string | null;
  status: CreativeTaskStatus;
  assigned_to_name?: string | null;
  depends_on_task_id?: string | null;
  started_at?: string | null;
  due_at?: string | null;
  completed_at?: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreativeTaskListResponse {
  data: CreativeTask[];
  meta: { count: number };
  error: null;
}

File: backend/app/schemas/creative_task.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.enums import CreativeTaskStatus
from app.schemas.common import TimestampedSchema


class CreativeTaskBase(BaseModel):
    design_request_id: UUID
    parent_task_id: UUID | None = None
    task_type: str
    title: str
    description: str | None = None
    status: CreativeTaskStatus
    assigned_to_name: str | None = None
    depends_on_task_id: UUID | None = None
    started_at: datetime | None = None
    due_at: datetime | None = None
    completed_at: datetime | None = None


class CreativeTaskCreate(CreativeTaskBase):
    pass


class CreativeTaskUpdate(BaseModel):
    status: CreativeTaskStatus | None = None
    assigned_to_name: str | None = None
    due_at: datetime | None = None
    completed_at: datetime | None = None


class CreativeTaskRead(CreativeTaskBase, TimestampedSchema):
    id: UUID


class CreativeTaskListResponse(BaseModel):
    data: list[CreativeTaskRead]
    meta: dict[str, int]
    error: None = None
